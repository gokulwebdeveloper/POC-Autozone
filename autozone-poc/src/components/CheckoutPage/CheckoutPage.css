.order-details-confirmation .card-header h6 {
  display: flex;
}
