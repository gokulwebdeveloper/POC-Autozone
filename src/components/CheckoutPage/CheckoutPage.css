.order-details-confirmation .card-header h6 {
  display: flex;
}

.order-details-form span{
  max-width: 80%;
}
