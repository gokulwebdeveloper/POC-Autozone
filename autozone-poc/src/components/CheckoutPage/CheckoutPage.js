import React from "react";
import BreadcrumbArea from "../BreadcrumbArea/BreadcrumbArea";
import InputWithLabel from "../../utlis/formComponents/InputWithLabel";
import Button from "../../utlis/button/Button";
import "./CheckoutPage.css";
import $ from 'jquery';

export default function CheckoutPage() {
  $(document).ready(function(){
    $('#checkoutBtn').on('click', function(){
      $('input[required]').each(function(element) {
        let val = $(this).val();
        if(val === ''){
          $('.checkout_area .formErrorMsg').text("please provide some input");
          $('.checkout_area .formErrorMsg:first').text("please fill all the fields");
          $('.checkout_area .formErrorMsg').show();
        }
      });
    })
  });
  return (
    <>
      <BreadcrumbArea title="Checkout" />

      <div className="checkout_area section-padding-80">
        <div className="container">
          <form action="#" method="post" className="row">
            <div className="col-12 col-md-6">
              <div className="checkout_details_area mt-50 clearfix">
                <div className="cart-page-heading mb-30">
                  <h5>Billing Address</h5>
                </div>
                <p className="formErrorMsg">Please fill all required fields</p>
                <div className="row">
                  <div className="col-md-6 mb-3">
                    <InputWithLabel
                      labelText="First Name"
                      inputId="first_name"
                      inputType="text"
                      isRequired={true}
                    />
                  </div>
                  <div className="col-md-6 mb-3">
                    <InputWithLabel
                      labelText="Last Name"
                      inputId="last_name"
                      inputType="text"
                      isRequired={true}
                    />
                  </div>
                  <div className="col-12 mb-3">
                    <InputWithLabel
                      labelText="Company Name"
                      inputId="company"
                      inputType="text"
                      isRequired={false}
                    />
                  </div>
                  <div className="col-12 mb-3">
                    <label htmlFor="country">
                      Country <span>*</span>
                    </label>
                    <select className="nice-select w-100" id="country">
                      <option value="usa">United States</option>
                      <option value="uk">United Kingdom</option>
                      <option value="ger">Germany</option>
                      <option value="fra">France</option>
                      <option value="ind">India</option>
                      <option value="aus">Australia</option>
                      <option value="bra">Brazil</option>
                      <option value="cana">Canada</option>
                    </select>
                  </div>
                  <div className="col-12 mb-3">
                    <InputWithLabel
                      labelText="Address"
                      inputId="street_address"
                      inputType="text"
                      isRequired={true}
                    />
                    <input
                      type="text"
                      className="form-control mt-3"
                      id="street_address2"
                    ></input>
                  </div>
                  <div className="col-12 mb-3">
                    <InputWithLabel
                      labelText="Postcode"
                      inputId="postcode"
                      inputType="text"
                      isRequired={true}
                    />
                  </div>
                  <div className="col-12 mb-3">
                    <InputWithLabel
                      labelText="Town/City"
                      inputId="city"
                      inputType="text"
                      isRequired={true}
                    />
                  </div>
                  <div className="col-12 mb-3">
                    <InputWithLabel
                      labelText="Province"
                      inputId="state"
                      inputType="text"
                      isRequired={true}
                    />
                  </div>
                  <div className="col-12 mb-3">
                    <InputWithLabel
                      labelText="Phone No"
                      inputId="phone_number"
                      inputType="text"
                      isRequired={true}
                      min="0"
                    />
                  </div>
                  <div className="col-12 mb-4">
                    <InputWithLabel
                      labelText="Email Address"
                      inputId="email_address"
                      inputType="email"
                      isRequired={true}
                    />
                  </div>

                  <div className="col-12">
                    <InputWithLabel
                      labelText="Terms and conitions"
                      inputId="customCheck1"
                      inputType="checkbox"
                      isRequired={false}
                    />
                    <InputWithLabel
                      labelText="Create an accout"
                      inputId="customCheck2"
                      inputType="checkbox"
                      isRequired={false}
                    />
                    <InputWithLabel
                      labelText="Subscribe to our newsletter"
                      inputId="customCheck3"
                      inputType="checkbox"
                      isRequired={false}
                    />
                  </div>
                </div>
              </div>
            </div>

            <div className="col-12 col-md-6 col-lg-5 ml-lg-auto">
              <div className="order-details-confirmation">
                <div className="cart-page-heading">
                  <h5>Your Order</h5>
                  <p>The Details</p>
                </div>

                <ul className="order-details-form mb-4">
                  <li>
                    <span>Product</span> <span>Total</span>
                  </li>
                  <li>
                    <span>Cocktail Yellow dress</span> <span>$59.90</span>
                  </li>
                  <li>
                    <span>Subtotal</span> <span>$59.90</span>
                  </li>
                  <li>
                    <span>Shipping</span> <span>Free</span>
                  </li>
                  <li>
                    <span>Total</span> <span>$59.90</span>
                  </li>
                </ul>

                <div id="accordion" role="tablist" className="mb-4">
                  <div className="card">
                    <div className="card-header" role="tab" id="headingOne">
                      <h6 className="mb-0">
                        <InputWithLabel
                          labelText="Paypal"
                          inputId="paymentRadio1"
                          inputType="radio"
                          isRequired={false}
                          name="paymentRadio"
                          defaultChecked
                        />
                        <a
                          className="collapsed"
                          data-toggle="collapse"
                          href="#collapseOne"
                          aria-expanded="false"
                          aria-controls="collapseOne"
                        >
                          <i className="fa fa-info-circle ml-2"></i>
                        </a>
                      </h6>
                    </div>

                    <div
                      id="collapseOne"
                      className="collapse"
                      role="tabpanel"
                      aria-labelledby="headingOne"
                      data-parent="#accordion"
                    >
                      <div className="card-body">
                        <p>
                          Lorem ipsum dolor sit amet, consectetur adipiscing
                          elit. Proin pharetra tempor so dales. Phasellus
                          sagittis auctor gravida. Integ er bibendum sodales
                          arcu id te mpus. Ut consectetur lacus.
                        </p>
                      </div>
                    </div>
                  </div>
                  <div className="card">
                    <div className="card-header" role="tab" id="headingTwo">
                      <h6 className="mb-0">
                        <InputWithLabel
                          labelText="cash on delievery"
                          inputId="paymentRadio2"
                          inputType="radio"
                          isRequired={false}
                          name="paymentRadio"
                        />
                        <a
                          className="collapsed"
                          data-toggle="collapse"
                          href="#collapseTwo"
                          aria-expanded="false"
                          aria-controls="collapseTwo"
                        >
                          <i className="fa fa-info-circle ml-2"></i>
                        </a>
                      </h6>
                    </div>
                    <div
                      id="collapseTwo"
                      className="collapse"
                      role="tabpanel"
                      aria-labelledby="headingTwo"
                      data-parent="#accordion"
                    >
                      <div className="card-body">
                        <p>
                          Lorem ipsum dolor sit amet, consectetur adipisicing
                          elit. Explicabo quis in veritatis officia inventore,
                          tempore provident dignissimos.
                        </p>
                      </div>
                    </div>
                  </div>
                  <div className="card">
                    <div className="card-header" role="tab" id="headingThree">
                      <h6 className="mb-0">
                        <InputWithLabel
                          labelText="credit card"
                          inputId="paymentRadio3"
                          inputType="radio"
                          isRequired={false}
                          name="paymentRadio"
                        />
                        <a
                          className="collapsed"
                          data-toggle="collapse"
                          href="#collapseThree"
                          aria-expanded="false"
                          aria-controls="collapseThree"
                        >
                          <i className="fa fa-info-circle ml-2"></i>
                        </a>
                      </h6>
                    </div>
                    <div
                      id="collapseThree"
                      className="collapse"
                      role="tabpanel"
                      aria-labelledby="headingThree"
                      data-parent="#accordion"
                    >
                      <div className="card-body">
                        <p>
                          Lorem ipsum dolor sit amet, consectetur adipisicing
                          elit. Esse quo sint repudiandae suscipit ab soluta
                          delectus voluptate, vero vitae
                        </p>
                      </div>
                    </div>
                  </div>
                  <div className="card">
                    <div className="card-header" role="tab" id="headingFour">
                      <h6 className="mb-0">
                        <InputWithLabel
                          labelText="direct bank transfer"
                          inputId="paymentRadio4"
                          inputType="radio"
                          isRequired={false}
                          name="paymentRadio"
                        />
                        <a
                          className="collapsed"
                          data-toggle="collapse"
                          href="#collapseFour"
                          aria-expanded="true"
                          aria-controls="collapseFour"
                        >
                          <i className="fa fa-info-circle ml-2"></i>
                        </a>
                      </h6>
                    </div>
                    <div
                      id="collapseFour"
                      className="collapse show"
                      role="tabpanel"
                      aria-labelledby="headingThree"
                      data-parent="#accordion"
                    >
                      <div className="card-body">
                        <p>
                          Lorem ipsum dolor sit amet, consectetur adipisicing
                          elit. Est cum autem eveniet saepe fugit, impedit
                          magni.
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <Button btnText="Place Order" id="checkoutBtn"/>
              </div>
            </div>
          </form>
        </div>
      </div>
    </>
  );
}
