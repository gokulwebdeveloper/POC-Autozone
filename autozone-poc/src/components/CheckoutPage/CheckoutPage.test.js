import React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import CheckoutPage from './CheckoutPage';

let page;
//mount component for testing
beforeEach(()=>{
    page = mount(<CheckoutPage />);
});

//Cleanup after test execution
afterEach(()=>{
    page.unmount();
});

describe("test CheckoutPage", ()=>{
    //test if CheckoutPage component exists
    it('CheckoutPage Exists', ()=>{
        expect(page.exists()).toBeTruthy();
    });

    //if it has title as "Billing address" and "Your Order"
    it('has titles as "Billing Address" and "Your Order"', ()=>{
        expect(page.find('h5').first().text()).toBe("Billing Address");
        expect(page.find('h5').last().text()).toBe("Your Order");
    });

    //if it has BreadcrumbArea componenent
    it('has BreadcrumbArea componenent', ()=>{
        expect(page.find('BreadcrumbArea')).toHaveLength(1);
    });

    //if it has 1 Button componenent
    it('has 1 Button componenent', ()=>{
        expect(page.find('Button')).toHaveLength(1);
    });

    //if Button has text Place Order
    it('Button has text Place Order', ()=>{
        expect(page.find('Button').text()).toBe("Place Order");
    });

    //if it has 1 form for order info
    it('has 1 form for order info', ()=>{
        expect(page.find('form')).toHaveLength(1);
    });

     //SnapShot Match test
     it("SnapShot Match", ()=>{
        const page = renderer.create(<CheckoutPage />).toJSON();
        expect(page).toMatchSnapshot()
    });
});