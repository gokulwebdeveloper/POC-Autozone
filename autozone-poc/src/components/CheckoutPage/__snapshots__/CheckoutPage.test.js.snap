// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CheckoutPage SnapShot Match 1`] = `
Array [
  <div
    className="breadcumb_area bg-img"
    style={
      Object {
        "backgroundImage": "url(breadcumb.jpg)",
      }
    }
  >
    <div
      className="container h-100"
    >
      <div
        className="row h-100 align-items-center"
      >
        <div
          className="col-12"
        >
          <div
            className="page-title text-center"
          >
            <h2>
              Checkout
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="checkout_area section-padding-80"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-md-6"
        >
          <div
            className="checkout_details_area mt-50 clearfix"
          >
            <div
              className="cart-page-heading mb-30"
            >
              <h5>
                Billing Address
              </h5>
            </div>
            <form
              action="#"
              method="post"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-6 mb-3"
                >
                  <label
                    htmlFor="first_name"
                  >
                    First Name
                     
                    <span>
                      *
                    </span>
                  </label>
                  <input
                    className="form-control"
                    defaultValue=""
                    id="first_name"
                    required={true}
                    type="text"
                  />
                </div>
                <div
                  className="col-md-6 mb-3"
                >
                  <label
                    htmlFor="last_name"
                  >
                    Last Name
                     
                    <span>
                      *
                    </span>
                  </label>
                  <input
                    className="form-control"
                    defaultValue=""
                    id="last_name"
                    required={true}
                    type="text"
                  />
                </div>
                <div
                  className="col-12 mb-3"
                >
                  <label
                    htmlFor="company"
                  >
                    Company Name
                     
                  </label>
                  <input
                    className="form-control"
                    defaultValue=""
                    id="company"
                    required={false}
                    type="text"
                  />
                </div>
                <div
                  className="col-12 mb-3"
                >
                  <label
                    htmlFor="country"
                  >
                    Country 
                    <span>
                      *
                    </span>
                  </label>
                  <select
                    className="nice-select w-100"
                    id="country"
                  >
                    <option
                      value="usa"
                    >
                      United States
                    </option>
                    <option
                      value="uk"
                    >
                      United Kingdom
                    </option>
                    <option
                      value="ger"
                    >
                      Germany
                    </option>
                    <option
                      value="fra"
                    >
                      France
                    </option>
                    <option
                      value="ind"
                    >
                      India
                    </option>
                    <option
                      value="aus"
                    >
                      Australia
                    </option>
                    <option
                      value="bra"
                    >
                      Brazil
                    </option>
                    <option
                      value="cana"
                    >
                      Canada
                    </option>
                  </select>
                </div>
                <div
                  className="col-12 mb-3"
                >
                  <label
                    htmlFor="street_address"
                  >
                    Address
                     
                    <span>
                      *
                    </span>
                  </label>
                  <input
                    className="form-control"
                    defaultValue=""
                    id="street_address"
                    required={true}
                    type="text"
                  />
                  <input
                    className="form-control mt-3"
                    id="street_address2"
                    type="text"
                  />
                </div>
                <div
                  className="col-12 mb-3"
                >
                  <label
                    htmlFor="postcode"
                  >
                    Postcode
                     
                    <span>
                      *
                    </span>
                  </label>
                  <input
                    className="form-control"
                    defaultValue=""
                    id="postcode"
                    required={true}
                    type="text"
                  />
                </div>
                <div
                  className="col-12 mb-3"
                >
                  <label
                    htmlFor="city"
                  >
                    Town/City
                     
                    <span>
                      *
                    </span>
                  </label>
                  <input
                    className="form-control"
                    defaultValue=""
                    id="city"
                    required={true}
                    type="text"
                  />
                </div>
                <div
                  className="col-12 mb-3"
                >
                  <label
                    htmlFor="state"
                  >
                    Province
                     
                    <span>
                      *
                    </span>
                  </label>
                  <input
                    className="form-control"
                    defaultValue=""
                    id="state"
                    required={true}
                    type="text"
                  />
                </div>
                <div
                  className="col-12 mb-3"
                >
                  <label
                    htmlFor="phone_number"
                  >
                    Phone No
                     
                    <span>
                      *
                    </span>
                  </label>
                  <input
                    className="form-control"
                    defaultValue=""
                    id="phone_number"
                    min="0"
                    required={true}
                    type="text"
                  />
                </div>
                <div
                  className="col-12 mb-4"
                >
                  <label
                    htmlFor="email_address"
                  >
                    Email Address
                     
                    <span>
                      *
                    </span>
                  </label>
                  <input
                    className="form-control"
                    defaultValue=""
                    id="email_address"
                    required={true}
                    type="email"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="custom-control custom-checkbox d-block mb-2"
                  >
                    <input
                      className="custom-control-input"
                      id="customCheck1"
                      required={false}
                      type="checkbox"
                    />
                    <label
                      className="custom-control-label"
                      htmlFor="customCheck1"
                    >
                      Terms and conitions
                       
                    </label>
                  </div>
                  <div
                    className="custom-control custom-checkbox d-block mb-2"
                  >
                    <input
                      className="custom-control-input"
                      id="customCheck2"
                      required={false}
                      type="checkbox"
                    />
                    <label
                      className="custom-control-label"
                      htmlFor="customCheck2"
                    >
                      Create an accout
                       
                    </label>
                  </div>
                  <div
                    className="custom-control custom-checkbox d-block mb-2"
                  >
                    <input
                      className="custom-control-input"
                      id="customCheck3"
                      required={false}
                      type="checkbox"
                    />
                    <label
                      className="custom-control-label"
                      htmlFor="customCheck3"
                    >
                      Subscribe to our newsletter
                       
                    </label>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          className="col-12 col-md-6 col-lg-5 ml-lg-auto"
        >
          <div
            className="order-details-confirmation"
          >
            <div
              className="cart-page-heading"
            >
              <h5>
                Your Order
              </h5>
              <p>
                The Details
              </p>
            </div>
            <ul
              className="order-details-form mb-4"
            >
              <li>
                <span>
                  Product
                </span>
                 
                <span>
                  Total
                </span>
              </li>
              <li>
                <span>
                  Cocktail Yellow dress
                </span>
                 
                <span>
                  $59.90
                </span>
              </li>
              <li>
                <span>
                  Subtotal
                </span>
                 
                <span>
                  $59.90
                </span>
              </li>
              <li>
                <span>
                  Shipping
                </span>
                 
                <span>
                  Free
                </span>
              </li>
              <li>
                <span>
                  Total
                </span>
                 
                <span>
                  $59.90
                </span>
              </li>
            </ul>
            <div
              className="mb-4"
              id="accordion"
              role="tablist"
            >
              <div
                className="card"
              >
                <div
                  className="card-header"
                  id="headingOne"
                  role="tab"
                >
                  <h6
                    className="mb-0"
                  >
                    <div
                      className="custom-control custom-radio d-block mb-2"
                    >
                      <input
                        className="custom-control-input"
                        defaultChecked={true}
                        id="paymentRadio1"
                        name="paymentRadio"
                        required={false}
                        type="radio"
                      />
                      <label
                        className="custom-control-label"
                        htmlFor="paymentRadio1"
                      >
                        Paypal
                         
                      </label>
                    </div>
                    <a
                      aria-controls="collapseOne"
                      aria-expanded="false"
                      className="collapsed"
                      data-toggle="collapse"
                      href="#collapseOne"
                    >
                      <i
                        className="fa fa-info-circle ml-2"
                      />
                    </a>
                  </h6>
                </div>
                <div
                  aria-labelledby="headingOne"
                  className="collapse"
                  data-parent="#accordion"
                  id="collapseOne"
                  role="tabpanel"
                >
                  <div
                    className="card-body"
                  >
                    <p>
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin pharetra tempor so dales. Phasellus sagittis auctor gravida. Integ er bibendum sodales arcu id te mpus. Ut consectetur lacus.
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  id="headingTwo"
                  role="tab"
                >
                  <h6
                    className="mb-0"
                  >
                    <div
                      className="custom-control custom-radio d-block mb-2"
                    >
                      <input
                        className="custom-control-input"
                        id="paymentRadio2"
                        name="paymentRadio"
                        required={false}
                        type="radio"
                      />
                      <label
                        className="custom-control-label"
                        htmlFor="paymentRadio2"
                      >
                        cash on delievery
                         
                      </label>
                    </div>
                    <a
                      aria-controls="collapseTwo"
                      aria-expanded="false"
                      className="collapsed"
                      data-toggle="collapse"
                      href="#collapseTwo"
                    >
                      <i
                        className="fa fa-info-circle ml-2"
                      />
                    </a>
                  </h6>
                </div>
                <div
                  aria-labelledby="headingTwo"
                  className="collapse"
                  data-parent="#accordion"
                  id="collapseTwo"
                  role="tabpanel"
                >
                  <div
                    className="card-body"
                  >
                    <p>
                      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Explicabo quis in veritatis officia inventore, tempore provident dignissimos.
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  id="headingThree"
                  role="tab"
                >
                  <h6
                    className="mb-0"
                  >
                    <div
                      className="custom-control custom-radio d-block mb-2"
                    >
                      <input
                        className="custom-control-input"
                        id="paymentRadio3"
                        name="paymentRadio"
                        required={false}
                        type="radio"
                      />
                      <label
                        className="custom-control-label"
                        htmlFor="paymentRadio3"
                      >
                        credit card
                         
                      </label>
                    </div>
                    <a
                      aria-controls="collapseThree"
                      aria-expanded="false"
                      className="collapsed"
                      data-toggle="collapse"
                      href="#collapseThree"
                    >
                      <i
                        className="fa fa-info-circle ml-2"
                      />
                    </a>
                  </h6>
                </div>
                <div
                  aria-labelledby="headingThree"
                  className="collapse"
                  data-parent="#accordion"
                  id="collapseThree"
                  role="tabpanel"
                >
                  <div
                    className="card-body"
                  >
                    <p>
                      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Esse quo sint repudiandae suscipit ab soluta delectus voluptate, vero vitae
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  id="headingFour"
                  role="tab"
                >
                  <h6
                    className="mb-0"
                  >
                    <div
                      className="custom-control custom-radio d-block mb-2"
                    >
                      <input
                        className="custom-control-input"
                        id="paymentRadio4"
                        name="paymentRadio"
                        required={false}
                        type="radio"
                      />
                      <label
                        className="custom-control-label"
                        htmlFor="paymentRadio4"
                      >
                        direct bank transfer
                         
                      </label>
                    </div>
                    <a
                      aria-controls="collapseFour"
                      aria-expanded="true"
                      className="collapsed"
                      data-toggle="collapse"
                      href="#collapseFour"
                    >
                      <i
                        className="fa fa-info-circle ml-2"
                      />
                    </a>
                  </h6>
                </div>
                <div
                  aria-labelledby="headingThree"
                  className="collapse show"
                  data-parent="#accordion"
                  id="collapseFour"
                  role="tabpanel"
                >
                  <div
                    className="card-body"
                  >
                    <p>
                      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Est cum autem eveniet saepe fugit, impedit magni.
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <button
              className="btn essence-btn"
            >
              Place Order
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
